compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/InteractionUIView.uikit.kt

```diff
@@ -16,38 +16,40 @@
 
 package androidx.compose.ui.window
 
+import androidx.compose.ui.viewinterop.InteropView
 import kotlinx.cinterop.CValue
 import kotlinx.cinterop.readValue
 import platform.CoreGraphics.CGPoint
 import platform.CoreGraphics.CGRectZero
 import platform.UIKit.UIEvent
 import platform.UIKit.UIPressesEvent
 import platform.UIKit.UIView
+import platform.UIKit.UITouchPhase
+import platform.UIKit.UIGestureRecognizer
 
-internal enum class UITouchesEventPhase {
+/**
+ * Subset of [UITouchPhase] reflecting immediate phase when event is received by the [UIView] or
+ * [UIGestureRecognizer].
+ */
+internal enum class CupertinoTouchesPhase {
     BEGAN, MOVED, ENDED, CANCELLED
 }
 
 internal class InteractionUIView(
-    private var keyboardEventHandler: KeyboardEventHandler,
-    private var touchesDelegate: Delegate,
-    private var updateTouchesCount: (count: Int) -> Unit,
+    private var hitTestInteropView: (point: CValue<CGPoint>, event: UIEvent?) -> InteropView?,
+    private var onTouchesEvent: (view: UIView, event: UIEvent, phase: CupertinoTouchesPhase) -> Unit,
+    private var onTouchesCountChange: (count: Int) -> Unit,
     private var inBounds: (CValue<CGPoint>) -> Boolean,
```

Comment:
I would consider some simplification of the API here:
- Move all calls to delegate-like interface.
- Omit `inBounds`, and move `hitTest` logic out to the class to its owner.

Reply:
I removed delegate specifically to avoid instantiation of objects that capture `this`.
Due to heavy coupling of dependencies in `ComposeSceneMediator` it's already a problem to figure out what's the order of `lazy` instantiations and I've tried this approach before I efficiently just made this. But I agree with the second point, we can make those things easier. Let's do it in a separate PR?